### ISSUE TITLE
new gd2 test errors

### ISSUE THREAD

**Issue by rurban** [lstein/Perl-GD#42]
@eserte has a fedora system with gd 2.3.2 and GD.t still fails with GD-2.74:.
2.73 works fine

```
#   Failed test 'unable to generate comparison image for test 1: Can't locate object method "newFromGd2" via package "GD::Image" at (eval 16) line 1.
# '
#   at t/GD.t line 259.
Can't locate object method "newFromGd2" via package "GD::Image" at t/GD.t line 80.
# Looks like your test exited with 255 just after 3.
t/GD.t .................... Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/14 subtests 
```

**Comment by eserte** [comment]
This can be reproduced using the following Dockerfile
```
FROM fedora:34
RUN yum -y install perl-CPAN gcc perl-Test perl-Test-Simple
RUN yum -y install gd-devel
RUN cpan -t GD
```
and
```
docker build -t perl-test . && docker run perl-test
```

Test output:
```
# Testing gd 2.3.2 using gd2 support.

#   Failed test 'unable to generate comparison image for test 1: Can't locate object method "newFromGd2" via package "GD::Image" at (eval 16) line 1.
# '
#   at t/GD.t line 259.
Can't locate object method "newFromGd2" via package "GD::Image" at t/GD.t line 80.
# Looks like your test exited with 255 just after 3.
t/GD.t .................... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/14 subtests 
```

**Comment by hakonhagland** [comment]
@eserte Thanks for providing the Dockerfile. I am able to reproduce the error with docker, and I can see that the error occurs at [line 94](https://github.com/lstein/Perl-GD/blob/master/t/GD.t#L94) : 
```
my $tile = eval "GD::Image->newFrom${op}('$fn')" or die $@;
```
where the value of `$op` is `Gd2` and the value of `$fn` is `./test_data/tile.gd2` (the file exists). However, the specific values of these variable seems to be unimportant. I tried replacing them with `Png` and `./test_data/tile.png` (the file exists) and I get the same error:

```
 Can't locate object method "newFromPng" via package "GD::Image"
```
I wonder if the error can have something to do with the installation location of the shared library:
```
$ ldd blib/arch/auto/GD/GD.so  | grep libgd
	libgd.so.3 => /lib64/libgd.so.3 (0x00007f9274326000)
```
However `cpan GD` uses the default path in `Makefile.PL`, se [line 203](https://github.com/lstein/Perl-GD/blob/master/Makefile.PL#L203) of `/usr/lib` instead of `lib64`

**Comment by rurban** [comment]
The image format selection in the test was still a hack. I'll change it the actual supported formats. Gif changed, GD changed and many formats depend if the support library was found or not.

**Comment by rurban** [comment]
Thanks for the reproducer!
I could fix it in devel, soon in 2.75

**Comment by rurban** [comment]
Merged
